lib/miq_automation_engine/engine/miq_ae_method_service/miq_ae_service_methods.rb

```diff
@@ -99,6 +99,21 @@ def self.tag_create(category, options = {})
       end
     end
 
+    def self.tag_delete(category, entry)
+      ar_method do
+        cat = Classification.find_by_name(category)
+        raise "Category <#{category}> does not exist" if cat.nil?
+
+        ent = cat.find_entry_by_name(entry)
+        raise "Entry <#{entry}> does not exist" if ent.nil?
+
+        raise "This tag has assignments. Please delete assignments before deleting the tag." if AssignmentMixin.all_assignments(ent.tag.name).present?
+
+        ent.destroy
+        true
```

Comment:
@astrozzc Is this true needed? Is it there to suppress the ent.destroy returning a ent. The true return indicates that their could be a false return too, which are exceptions.

Reply:
@astrozzc You might want to explore the ! methods in rails which raise an exception otherwise they might return false and suppress the exception. save!, create! etc.

Reply:
Yes, it is to suppress the return of ent obj. I add it to make it consistent with the create method. If there is something wrong with the .destroy, then the return would be false or exception.

Reply:
Looks like the solution here is to use `destroy!` to ensure the method is successful or raises an error.

Reply:
Thanks for the suggestion!

Reply:
Ruby allows a simpler syntax here:

```ruby
def self.tag_delete(category, entry)
  tag_delete!(category, entry)
  true
rescue
  false
end
```